### PR TITLE
Simplify, generalize and improve GaussianBlur

### LINKED ISSUE #208: GaussianBlur kernel tiling may cause cross-channel mixing instead of per-channel blur
## Issue Description

During code review of PR #204, a potential issue was identified in the `GaussianBlur` operation implementation where the kernel tiling approach may cause cross-channel mixing instead of independent per-channel Gaussian blur.

## Location
- File: `zea/ops/tensor.py`
- Lines: 83-94
- Operation: `GaussianBlur.call()`

## Context
The current implementation tiles the kernel to shape `(k, k, C, C)` and uses `ops.conv`, which creates a full convolution where each output channel is a weighted sum of all input channels. This may not be the intended behavior for a Gaussian blur, which typically applies the blur independently to each channel.

## References
- PR: https://github.com/tue-bmd/zea/pull/204
- Comment: https://github.com/tue-bmd/zea/pull/204#discussion_r2610642527
- Requested by: @tristan-deep

## Next Steps
Investigate whether the current behavior is intentional or if a depthwise/per-channel convolution approach would be more appropriate.

### PR DESCRIPTION
**Pull request by wesselvannierop**
Uses `zea.func.gaussian_filter` now. Fixes #208 as a result.

## Summary by CodeRabbit

* **Bug Fixes**
  * Improved numerical precision for blur/denoise comparisons with tighter tolerances.

* **Tests**
  * Multi-backend tests narrowed to compatible backends and now return computed outputs for validation.

* **Documentation**
  * Corrected TensorFlow reference in docstrings.

* **Refactor**
  * Simplified blur and denoise filter interfaces with consolidated parameters (order, mode, cval, truncate, axes) and streamlined axis handling for more consistent behavior.

### CODE REVIEW

File: tests/test_operations.py [hunk 1]
```diff
@@ -425,15 +431,19 @@ def test_lee_filter(sigma, spiral_image):
...
+    filtered_batched = keras.ops.convert_to_numpy(filtered_batched)
+
+    assert np.allclose(filtered, filtered_batched), (
+        "LeeFilter with and without batch dim should give the same result."
+    )
 
-    assert keras.ops.var(filtered) < keras.ops.var(image_tensor), (
+    assert np.var(filtered) < np.var(image_tensor), (
         "LeeFilter should reduce variance of the processed image"
     )
```
[the remark this answers is not in the document]
Comment (tristan-deep) [reply tests/test_operations.py]: I agree actually probably best practice to convert image tensor to numpy first. You don't test with torch backend, but that would complain if you do not convert explicitely to numpy.

**Review by tristan-deep — APPROVED** [review]
Nice! Like that you are now using the `zea.func.tensor.gaussian_filter` inside the ops.
